src/load_med.cpp

```diff
@@ -614,39 +614,40 @@ BOOL CSoundFile::ReadMed(const BYTE *lpStream, DWORD dwMemLength)
 	}
 	m_nDefaultTempo = deftempo;
 	// Reading Samples
-	for (UINT iSHdr=0; iSHdr<m_nSamples; iSHdr++)
+	psample = pmsh;
+	for (UINT iSHdr=0; iSHdr<m_nSamples; iSHdr++, psample += sizeof(MMD0SAMPLE))
 	{
 		MODINSTRUMENT *pins = &Ins[iSHdr+1];
-		pins->nLoopStart = bswapBE16(pmsh->sample[iSHdr].rep) << 1;
-		pins->nLoopEnd = pins->nLoopStart + (bswapBE16(pmsh->sample[iSHdr].replen) << 1);
-		pins->nVolume = (pmsh->sample[iSHdr].svol << 2);
+		pins->nLoopStart = READ_BE16(psample + offsetof(MMD0SAMPLE,rep)) << 1;
+		pins->nLoopEnd = pins->nLoopStart + (READ_BE16(psample + offsetof(MMD0SAMPLE,replen)) << 1);
+		pins->nVolume = *(psample + offsetof(MMD0SAMPLE,svol)) << 2;
 		pins->nGlobalVol = 64;
 		if (pins->nVolume > 256) pins->nVolume = 256;
-		pins->RelativeTone = -12 * pmsh->sample[iSHdr].strans;
+		pins->RelativeTone = -12 * (signed char) *(psample + offsetof(MMD0SAMPLE,strans));
 		pins->nPan = 128;
 		if (pins->nLoopEnd) pins->uFlags |= CHN_LOOP;
 	}
 	// Common Flags
-	if (!(pmsh->flags & 0x20)) m_dwSongFlags |= SONG_FASTVOLSLIDES;
+	if (!(flags & 0x20)) m_dwSongFlags |= SONG_FASTVOLSLIDES;
 	// Reading play sequence
 	if (version < '2')
 	{
-		UINT nbo = pmsh->songlen >> 8;
+		UINT nbo = READ_BE16(pmsh + offsetof(MMD0SONGHEADER,songlen));
```

Comment:
this isn't a change request, but I felt like pointing out the old version here broke my brain a little bit :(

Reply:
Yeah. There are some other similar hacks there, all changed to use `READ_BE??`